tests/large-lib-test.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+from testUtils import Utils
+from Cluster import Cluster
+from WalletMgr import WalletMgr
+from Node import Node
+from Node import ReturnType
+from TestHelper import TestHelper
+
+import random
+import signal
+import shutil
+import time
+
+###############################################################
+# large-lib-test
+#
+# Test LIB in a network will advance when an invalid larger LIB 
+# than current one is received from a speculative node.
+#
+###############################################################
+
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+args=TestHelper.parse_args({"--kill-sig","--kill-count","--keep-logs"
+                            ,"--dump-error-details","-v","--leave-running","--clean-run"
+                            })
+pnodes=1
+total_nodes=3 # first one is producer, and last two are speculative nodes
+debug=args.v
+killEosInstances=not args.leave_running
+dumpErrorDetails=args.dump_error_details
+keepLogs=args.keep_logs
+killAll=args.clean_run
+relaunchTimeout=10
+numBlocksToProduceBeforeRelaunch=80 # Don't want to set too big, trying to reduce test time
+secsToWaitBeforeChecking=20
+
+Utils.Debug=debug
+testSuccessful=False
+
+seed=1
+random.seed(seed) # Use a fixed seed for repeatability.
+cluster=Cluster(walletd=True)
+walletMgr=WalletMgr(True)
+cluster.setWalletMgr(walletMgr)
+
+def relaunchNode(node: Node, chainArg="", skipGenesis=True, relaunchAssertMessage="Fail to relaunch"):
+   isRelaunchSuccess=node.relaunch(chainArg=chainArg, timeout=relaunchTimeout, skipGenesis=skipGenesis, cachePopen=True)
+   time.sleep(1) # Give a second to replay or resync if needed
+   assert isRelaunchSuccess, relaunchAssertMessage
+   return isRelaunchSuccess
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+
+    cluster.killall(allInstances=killAll)
+    cluster.cleanup()
+    walletMgr.killall(allInstances=killAll)
+    walletMgr.cleanup()
+
+    # set the last two nodes as speculative
+    specificExtraNodeosArgs={}
+    specificExtraNodeosArgs[1]="--read-mode speculative "
+    specificExtraNodeosArgs[2]="--read-mode speculative "
+
+    Print("Stand up cluster")
+    if cluster.launch(
+            pnodes=pnodes,
+            totalNodes=total_nodes,
+            totalProducers=1,
+            useBiosBootFile=False,
+            topo="mesh",
```

Comment:
FYI, mesh is default

Reply:
I know, but I like it being explicit since it is important to the test.